### ISSUE TITLE
'Info classification' requirement for HNAP2.3

### ISSUE THREAD

**Issue by alexandreleroux** [geopython/pygeometa#20]
HNAP2.3 has mandatory requirements for 'Information classification'.  There are 5 parameters:
- Information Category: mandatory
- Geography: mandatory
- Content: optional, can have multiple values
- Program: optional, can have multiple values
- GC Security Level: mandatory

ECDC provides dropdown lists of value values for these parameters. In the pygeometa spirit, I suggest specifying 'Information classification' directly in the MCF file. Parameter names could be HNAP_info-cat, HNAP_geography, HNAP_info-content, HNAP_info-program and HNAP_GCSecLevel

Example of output XML which has 'Information classification' specified in it. For Program and GC Security Level, the associated code goes in the same section EC_CorporateInfo mentioned in issue #19 :

```
                                <napec:GC_Security_Classification>
                                        <napec:GC_Security_Classification_TypeCode codeListValue="unclassified" codeList="http://www.ec.gc.ca/data_donnees/standards/schemas/napec#GC_Security_Classification"/>
                                </napec:GC_Security_Classification>
                                <napec:EC_Program>
                                        <napec:EC_Program_TypeCode codeListValue="weather_observations_forecasts_and_warnings" codeList="http://www.ec.gc.ca/data_donnees/standards/schemas/napec#EC_Program"/>
                                </napec:EC_Program>
                        </napec:EC_CorporateInfo>
                </napec:MD_DataIdentification>
```

For info-cat, geography and content, they go in the gmd:descriptivekeywords section, such as:

```
                        <gmd:descriptiveKeywords>
                                <gmd:MD_Keywords>
                                        <gmd:keyword xmlns:srv="http://www.isotc211.org/2005/srv" xsi:type="gmd:PT_FreeText_PropertyType">
                                                <gco:CharacterString>Weather and Climate</gco:CharacterString>
                                                <gmd:PT_FreeText>
                                                        <gmd:textGroup>
                                                                <gmd:LocalisedCharacterString locale="#fra">Temps et climat</gmd:LocalisedCharacterString>
                                                        </gmd:textGroup>
                                                </gmd:PT_FreeText>
                                        </gmd:keyword>
                                        <gmd:type xmlns:srv="http://www.isotc211.org/2005/srv">
                                                <gmd:MD_KeywordTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_101" codeListValue="theme"/>
                                        </gmd:type>
                                        <gmd:thesaurusName>
                                                <gmd:CI_Citation id="local.theme.EC_Information_Category">
                                                        <gmd:title>
                                                                <gco:CharacterString>local.theme.EC_Information_Category</gco:CharacterString>
                                                        </gmd:title>
                                                        <gmd:date>
                                                                <gmd:CI_Date>
                                                                        <gmd:date>
                                                                                <gco:Date>2012-05-25</gco:Date>
                                                                        </gmd:date>
                                                                        <gmd:dateType xmlns:srv="http://www.isotc211.org/2005/srv">
                                                                                <gmd:CI_DateTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_87" codeListValue="RI_367">publication; publication</gmd:CI_DateTypeCode>
                                                                        </gmd:dateType>
                                                                </gmd:CI_Date>
                                                        </gmd:date>
                                                        <gmd:citedResponsibleParty/>
                                                </gmd:CI_Citation>
                                        </gmd:thesaurusName>
                                </gmd:MD_Keywords>
                        </gmd:descriptiveKeywords>
                        <gmd:descriptiveKeywords>
                                <gmd:MD_Keywords>
                                        <gmd:keyword xmlns:srv="http://www.isotc211.org/2005/srv" xsi:type="gmd:PT_FreeText_PropertyType">
                                                <gco:CharacterString>International</gco:CharacterString>
                                                <gmd:PT_FreeText>
                                                        <gmd:textGroup>
                                                                <gmd:LocalisedCharacterString locale="#fra">Internationale</gmd:LocalisedCharacterString>
                                                        </gmd:textGroup>
                                                </gmd:PT_FreeText>
                                        </gmd:keyword>
                                        <gmd:type xmlns:srv="http://www.isotc211.org/2005/srv">
                                                <gmd:MD_KeywordTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_101" codeListValue="place"/>
                                        </gmd:type>
                                        <gmd:thesaurusName>
                                                <gmd:CI_Citation id="local.place.EC_Geographic_Scope">
                                                        <gmd:title>
                                                                <gco:CharacterString>local.place.EC_Geographic_Scope</gco:CharacterString>
                                                        </gmd:title>
                                                        <gmd:date>
                                                                <gmd:CI_Date>
                                                                        <gmd:date>
                                                                                <gco:Date>2012-05-25</gco:Date>
                                                                        </gmd:date>
                                                                        <gmd:dateType xmlns:srv="http://www.isotc211.org/2005/srv">
                                                                                <gmd:CI_DateTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_87" codeListValue="RI_367">publication; publication</gmd:CI_DateTypeCode>
                                                                        </gmd:dateType>
                                                                </gmd:CI_Date>
                                                        </gmd:date>
                                                        <gmd:citedResponsibleParty/>
                                                </gmd:CI_Citation>
                                        </gmd:thesaurusName>
                                </gmd:MD_Keywords>
                        </gmd:descriptiveKeywords>
                        <gmd:descriptiveKeywords>
                                <gmd:MD_Keywords>
                                        <gmd:keyword xmlns:srv="http://www.isotc211.org/2005/srv" xsi:type="gmd:PT_FreeText_PropertyType">
                                                <gco:CharacterString>Environmental Model</gco:CharacterString>
                                                <gmd:PT_FreeText>
                                                        <gmd:textGroup>
                                                                <gmd:LocalisedCharacterString locale="#fra">Modèle environnemental</gmd:LocalisedCharacterString>
                                                        </gmd:textGroup>
                                                </gmd:PT_FreeText>
                                        </gmd:keyword>
                                        <gmd:type xmlns:srv="http://www.isotc211.org/2005/srv">
                                                <gmd:MD_KeywordTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_101" codeListValue="theme"/>
                                        </gmd:type>
                                        <gmd:thesaurusName>
                                                <gmd:CI_Citation id="local.theme.EC_Content_Scope">
                                                        <gmd:title>
                                                                <gco:CharacterString>local.theme.EC_Content_Scope</gco:CharacterString>
                                                        </gmd:title>
                                                        <gmd:date>
                                                                <gmd:CI_Date>
                                                                        <gmd:date>
                                                                                <gco:Date>2012-05-25</gco:Date>
                                                                        </gmd:date>
                                                                        <gmd:dateType xmlns:srv="http://www.isotc211.org/2005/srv">
                                                                                <gmd:CI_DateTypeCode codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_87" codeListValue="RI_367">publication; publication</gmd:CI_DateTypeCode>
                                                                        </gmd:dateType>
                                                                </gmd:CI_Date>
                                                        </gmd:date>
                                                        <gmd:citedResponsibleParty/>
                                                </gmd:CI_Citation>
                                        </gmd:thesaurusName>
                                </gmd:MD_Keywords>
                        </gmd:descriptiveKeywords>
```

Let me know if you need additional info. Thanks! -- Alex

**Comment by tomkralidis** [comment]
Information Category: mandatory
    Geography: mandatory
    Content: optional, can have multiple values
    Program: optional, can have multiple values
    GC Security Level: mandatory

1. for program and GC Security Level as discussed in #19 you can use an XML injection as a post-process for `napec` based elements
1. for the information category, geography and content classifications how about 3 more hnap-specific MCF parameters?
 - `hnap_category_information_en`, `hnap_category_information_fr`
 - `hnap_category_geography_en`, `hnap_category_geography_fr`
 - `hnap_category_content_en`, `hnap_category_content_fr`

**Comment by alexandreleroux** [comment]
@tomkralidis 
1. Agreed, GC Security Level can be dealt with via post-pygeometa xml injection just like issue #19, I'll try to take care of this 
2. Ok for the new hnap-specific parameters for info+geo+content. Since these will have different values based on the dataset, it needs to be specified in the MCF itself, I'm letting you implement that part

**Comment by tomkralidis** [comment]
@alexandreleroux implemented in master, can you verify?

**Comment by alexandreleroux** [comment]
@tomkralidis thanks, after fixing typos, I confirms it works.

Leaving ticket open to address:
1. GC Security Level
2. Update pygeometa's hnap documentation for the 3 new hnap_category_* parameters

**Comment by alexandreleroux** [comment]
gc security level addressed by xml injection implemented for #19 . Need to update documentation accordingly

**Comment by alexandreleroux** [comment]
After validation, this is a requirement on top of the HNAP schema and should be handled outside of pygeometa. Closing
